app/src/main/java/ch/epfl/sdp/drone3d/service/api/drone/DronePhotos.kt

```diff
@@ -0,0 +1,47 @@
+package ch.epfl.sdp.drone3d.service.api.drone
+
+import android.graphics.Bitmap
+import androidx.lifecycle.LiveData
+import io.reactivex.Single
+
+/**
+ * An interface that allows you get photos taken by the drone
+ */
+interface DronePhotos {
+
+    /**
+     * Returns a live data containing a bitmap of the last image taken.
+     */
+    fun getNewPhotos(): LiveData<Bitmap>
+
+    /**
+     * Returns all the image of the drone since its connection.
+     * It may not return all photos in case a download failed.
+     */
+    fun getPhotos(): Single<List<Bitmap>>
+
+    /**
+     * Returns the last [n] images taken by the drone.
+     * If [n] is negative, it will return an empty list.
+     * If the drone took less than [n] images, it will return all of them.
+     * It may return less than [n] photos in case a download failed.
+     */
+    fun getLastPhotos(n: Int): Single<List<Bitmap>>
+
+    /**
+     * Returns the first [n] images taken by the drone.
+     * If [n] is negative, it will return an empty list.
+     * If the drone took less than [n] images, it will return all of them.
+     * It may return less than [n] photos in case a download failed.
+     */
+    fun getFirstPhotos(n: Int): Single<List<Bitmap>>
+
+    /**
+     * Returns [n] random images taken by the drone.
+     * If [n] is negative, it will return an empty list.
+     * If the drone took less than [n] images, it will return all of them.
+     * It may return less than [n] photos in case a download failed.
+     */
+    fun getRandomPhotos(n: Int): Single<List<Bitmap>>
+
```

Comment:
Would it be possible to create a function that returns the urls of all Photos ?

So that it doesn't have to be downloaded on the phone and can be downloaded manually by the user.

Reply:
Yes I just added the getPhotosUrl function